examples/conanfile/package_id/settings_and_options.rst

```diff
@@ -0,0 +1,206 @@
+                      compiler.version: 193
+                      os: Windows
+
+However, there are cases when a setting should not influence the resulting package ID.
+An example of this could be when a project generates an executable binary as a compilation result. In that case, the value of the compiler
+used is not relevant for the package and we can remove its influence in the package ID like so:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        settings = "os", "arch", "compiler", "build_type"
+
+        def build(self):
+            # self.settings.compiler value will be used here to compile
+
+        def package_id(self):
+            # Remove compiler setting from package ID
+            del self.info.settings.compiler
+
+Why not removing the setting from the `settings` attribute? Because the compiler value is still needed in the `build()` method to perform the compilation of the executable.
+
+
+How options influence the package ID
+-------------------------------------
+
+Options are used to specify characteristics that are particular to a single recipe, contrasting with settings that generally remain consistent across recipes within a project.
+They are usually a set of particular characteristic a library executble or conan package may have.
+
+For example, a `shared` option is a very common option used in recipes that can produce shared libraries. However, it could not be a setting as not all recipes produce shared libraries.
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        options = {"shared": [True, False]}
+        default_options = {"shared": True}
+
+As in the previous case with settings, the different values of an option will influence the package ID and therefore, generate different packages depending on it.
+
+.. code-block:: bash
+   :emphasize-lines: 3, 7
+
+    $ conan create . --options shared=True
+    ...
+    pkg/1.0.0: Package '1744785cb24e3bdca70e27041dc5abd20476f947' created
+
+    $ conan create . --options shared=False
+    ...
+    pkg/1.0.0: Package '55c609fe8808aa5308134cb5989d23d3caffccf2' created
+
+In the same way, there might be options that are needed as input in a recipe to generate a package which shouldn't be taken into account in the package ID.
+To do so, the recipe can remove the option in the :ref:`package_id() method <reference_conanfile_methods_package_id>`:
+
+.. code-block:: python
+
+    from conan import ConanFile
+
+    class Pkg(ConanFile):
+        name = "pkg"
+        version = "1.0.0"
+        options = {"diagnostics_level": [0, 1, 2]}
+        default_options = {"diagnostics_level": 0}
+
+        def build(self):
+            # self.options.diagnostics_level value will be used here
+
+        def package_id(self):
+            # Remove diagnostics_level option from package ID
```

Comment:
I think this direct removal of the option is still misleading. If the options are being used in the ``build()`` how is it that its not producing a different binary? And if it is not used at ``build()``, then why not removing it earlier in ``config_options()`` or ``configure()``?

I think I'd still prefer the opposite, the lines below are not that evident or explicit, and it would be better to say: "if you have options and you want to remove them from package_id(), then probably they are not options, so better use ``conf``", instead of just putting the code here, which looks like a legitimate, recommended and blessed usage.

Reply:
I can think of 3 patterns for options (there may be more):
1. The ones that affect what is being built (so they should affect package ID) - this is the most common case
2. The ones that don't affect the actual product, but control something that only makes sense during the `build()` phase - think the verbosity of the boost build configurator, or "verbose makefile" or something of the likes - I think that's the example being shown here with `diagnostics_level` - perhaps it could
3. options that don't affect either (1) or (2), but affect consumers (think `INTERFACE` type options in CMake targets). There's a few of these cases. These would typically be evaluated only in the `package_id()` method.

> If the options are being used in the build() how is it that its not producing a different binary?

@memsharded  I think the example being shown here would be what I'm listing as (2) - perhaps it could be clearer.

Reply:
Yes, the approach presented here is 2). My main concern is that this shouldn't be presented this way, looking like an almost standard mechanism of passing information to the recipes without affecting their binaries. 

By defaults ``options`` should be binary-affecting options, and for things that are not binary-affecting, then the general recommended way is ``conf``, not options. This is not clear and evident from this writing.

Reply:
Thanks for the comments. Indeed, I agree with you and I don't want to create any controversy here and just expose a clear usage to the user.

If we think this isn't a very common thing for managing options, we can just drop the example. I just wanted to show how options/settings influence the package ID and removing them in the package_id() method was a good example for me to show the user the influence on the result (not only tied to the input values of the options/settings).

Anyway, I'd rather remove it than add complexity with the other examples or cases 👍

Reply:
Yes, lets elaborate a bit more instead of presenting this case, as is. Please describe the 2 strategies:
- Using an ``option`` and remove it (in text, no code). In principle ``options`` are there to affect the ``package_id``, so in general ``conf`` might be preferred. 
- Using a ``conf``. Recommended for general configuration, that end user can select, possibly common for more than one package